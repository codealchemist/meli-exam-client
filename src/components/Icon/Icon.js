import React from 'react'
import styled from 'styled-components'
import Colors from '../Colors'

const Wrapper = styled.div`
  width: 46px;
  height: 39px;
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: 21px;
  color: ${Colors.default.dark2};
  background: ${props => props.background || 'transparent'};

  i {
    font-style: normal;
    width: 100%;
    text-align: center;
    margin-top: 4px;
  }
`

const Icon = ({ type, background, onClick }) => (
  <Wrapper background={background} onClick={onClick}>
    <i className={type} />
  </Wrapper>
)

export default Icon
