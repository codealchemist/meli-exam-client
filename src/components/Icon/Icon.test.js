import React from 'react'
import { mount } from 'enzyme'
import Icon from './Icon'

describe('Icon', () => {
  it('should render', () => {
    const wrapper = mount(<Icon />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with type', () => {
    const type = 'rock'
    const wrapper = mount(<Icon type={type} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with custom background', () => {
    const background = 'black'
    const wrapper = mount(<Icon background={background} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass onClick', () => {
    const onClick = jest.fn()
    const wrapper = mount(<Icon onClick={onClick} />)
    expect(wrapper).toMatchSnapshot()

    wrapper.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
