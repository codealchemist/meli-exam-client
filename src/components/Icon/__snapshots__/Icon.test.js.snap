// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should pass onClick 1`] = `
<Icon
  onClick={[MockFunction]}
>
  <styled.div
    onClick={[MockFunction]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bPZTKg",
            "rules": Array [
              "
  width: 46px;
  height: 39px;
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: 21px;
  color: ",
              "#666",
              ";
  background: ",
              [Function],
              ";

  i {
    font-style: normal;
    width: 100%;
    text-align: center;
    margin-top: 4px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[MockFunction]}
    >
      <div
        className="sc-bdVaJa bPZTKg"
        onClick={[MockFunction]}
      >
        <i />
      </div>
    </StyledComponent>
  </styled.div>
</Icon>
`;

exports[`Icon should render 1`] = `
<Icon>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bPZTKg",
            "rules": Array [
              "
  width: 46px;
  height: 39px;
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: 21px;
  color: ",
              "#666",
              ";
  background: ",
              [Function],
              ";

  i {
    font-style: normal;
    width: 100%;
    text-align: center;
    margin-top: 4px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa bPZTKg"
      >
        <i />
      </div>
    </StyledComponent>
  </styled.div>
</Icon>
`;

exports[`Icon should render with custom background 1`] = `
<Icon
  background="black"
>
  <styled.div
    background="black"
  >
    <StyledComponent
      background="black"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fspGsS",
            "rules": Array [
              "
  width: 46px;
  height: 39px;
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: 21px;
  color: ",
              "#666",
              ";
  background: ",
              [Function],
              ";

  i {
    font-style: normal;
    width: 100%;
    text-align: center;
    margin-top: 4px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa fspGsS"
      >
        <i />
      </div>
    </StyledComponent>
  </styled.div>
</Icon>
`;

exports[`Icon should render with type 1`] = `
<Icon
  type="rock"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bPZTKg",
            "rules": Array [
              "
  width: 46px;
  height: 39px;
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: 21px;
  color: ",
              "#666",
              ";
  background: ",
              [Function],
              ";

  i {
    font-style: normal;
    width: 100%;
    text-align: center;
    margin-top: 4px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa bPZTKg"
      >
        <i
          className="rock"
        />
      </div>
    </StyledComponent>
  </styled.div>
</Icon>
`;
